tools/gengen/util/generator.go

```diff
@@ -526,7 +526,11 @@ func (g *GenesisGenerator) setupPost(ctx context.Context, addr, mIDAddr address.
 	}
 
 	// set provingperiod start state field
-	minerActorState.PoStState.ProvingPeriodStart = miner.ProvingPeriod
+	ppStart := abi.ChainEpoch(miner.ProvingPeriod)
+	if provingPeriodStart != nil {
```

Comment:
Minor preference for using 0 or -1 for the sentinel rather than introducing nil pointer possibilities here.